examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,271 @@
+:math:`\beta = \frac{1}{\sqrt{2}} \int \sqrt{P(t)} dt` 
+"""
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+import numpy as np
+from itertools import cycle
+
+colorcyc_list = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+color_cycle = cycle(
+    colorcyc_list
+)  # this can be done more than once to spin up multiple lists
+
+V_atten_ratio = 102.2  # attenutation ratio
+skip_plots = 33  # diagnostic -- set this to None, and there will be no plots
+slicewidth = 1e6
+typical_180 = 40e-6  # typical beta for a 180 -- it's really important to get pulses in this regime correct
+
+
+with psd.figlist_var() as fl:
+    for filename, nodename, linear_threshold in [
+        (
+            "240819_test_amp0p05_calib_pulse_calib.h5",
+            "pulse_calib_3",
+            150e-6,
+        ),  # low power
+        (
+            "240819_amp0p1_calib_pulse_calib.h5",
+            "pulse_calib_1",
+            270e-6,
+        ),  # low power
+        (
+            "240819_amp0p2_calib_repeat_pulse_calib.h5",
+            "pulse_calib_8",
+            310e-6,
+        ),  # low power
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        assert (
+            d.get_prop("postproc_type") == "GDS_capture_v1"
+        ), "The wrong postproc_type was set so you most likely used the wrong script for acquisition"
+        amplitude = d.get_prop("acq_params")["amplitude"]
+        fl.basename = f"amplitude = {amplitude}"
+        if not d.get_units("t") == "s":
+            print(
+                "units weren't set for the t axis or else I can't read them from the hdf5 file!"
+            )
+            d.set_units("t", "s")
+        d *= V_atten_ratio
+        d /= np.sqrt(50)  # V/sqrt(R) = sqrt(P)
+
+        def switch_to_plot(d, j):
+            thislen = d["t_pulse"][j]
+            fl.next(f"pulse length = {thislen}")
+
+        def indiv_plots(d, thislabel, thiscolor):
+            if skip_plots is None:
+                return
+            for j in range(len(d["t_pulse"])):
+                if j % skip_plots == 0:
+                    switch_to_plot(d, j)
+                    fl.plot(
+                        d["t_pulse", j],
+                        alpha=0.2,
+                        color=thiscolor,
+                        label=thislabel,
+                    )
+
+        indiv_plots(d, "raw", "blue")
+        # {{{ data is already analytic, and downsampled to below 24 MHz
+        indiv_plots(abs(d), "analytic", "orange")
+        d.ft("t")
+        # {{{ apply frequency filter
+        d.ift("t")
+        SW = 1 / (d["t"][1] - d["t"][0])  # sample rate
+        carrier = (
+            d.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+        )  # signal frequency
+        n = np.floor(
+            (carrier + SW / 2) / SW
+        )  # nearest integer multiple of sampling frequency
+        nu_a = carrier - n * SW  # aliasing frequency
```

Comment:
It would seem to me that if this is negative, you would want -carrier plus an integer SW instead